### ISSUE TITLE
RPC Error: Transport is closing

### ISSUE THREAD

**Issue by jbertman** [cloudquery/cq-provider-azure#10]
As of this morning, with a freshly generated configuration (with `subscriptions` commented out), I can't seem to run the fetching process:

```
Initializing CloudQuery Providers...

✓ cq-provider-azure@v0.2.1 verified  0s  100 %

Finished provider initialization...

Starting provider fetch...

⌛cq-provider-azure@latest fetching  1s [=====================================================>--------|  Finished Resources: 7/8
Error: rpc error: code = Unavailable desc = transport is closing
```

I have tried clean database instances and pulling the latest cloudquery code. Any insight?

**Comment by roneli** [comment]
Will have a look at this, I am assuming fetched all resources? might be a specific resource causing this problem.

**Comment by jbertman** [comment]
@roneli Thanks for the quick turnaround. I'm wondering if maybe it's a permission issue? I'm not seeing any errors in either the console log or the cloudquery.log:

```
11:52AM INF logging configured consoleLog=true fileLogging=true fileName=cloudquery.log jsonLogOutput=false logDirectory=. maxAgeInDays=3 maxBackups=3 maxSizeMB=30 verbose=false
11:52AM INF Initializing required providers
11:52AM INF Initializing provider name=azure version=latest
11:52AM INF requesting provider to configure provider=azure version=v0.2.1
11:52AM INF No subscriptions specified going to using all available ones @module=azure subscriptions=["<SNIP>"] timestamp=2021-06-08T11:52:36.486-0400
11:52AM INF provider configured successfully provider=azure version=v0.2.1
11:52AM INF provider started fetching resources provider=azure version=v0.2.1
11:52AM INF fetched successfully @module=azure count=0 subscription_id=<SNIP> table=azure_keyvault_vaults timestamp=2021-06-08T11:52:36.559-0400
11:52AM INF fetched successfully @module=azure count=0 subscription_id=<SNIP> table=azure_mysql_servers timestamp=2021-06-08T11:52:36.608-0400
11:52AM INF fetched successfully @module=azure count=0 subscription_id=<SNIP> table=azure_postgresql_servers timestamp=2021-06-08T11:52:36.660-0400
11:52AM INF fetched successfully @module=azure count=0 subscription_id=<SNIP> table=azure_sql_servers timestamp=2021-06-08T11:52:36.715-0400
11:52AM INF fetched successfully @module=azure count=2 subscription_id=<SNIP> table=azure_resource_groups timestamp=2021-06-08T11:52:36.750-0400
11:52AM INF fetched successfully @module=azure count=0 subscription_id=<SNIP> table=azure_storage_accounts timestamp=2021-06-08T11:52:36.750-0400
11:52AM INF fetched successfully @module=azure count=1 subscription_id=<SNIP> table=azure_compute_disks timestamp=2021-06-08T11:52:36.793-0400
Error: rpc error: code = Unavailable desc = transport is closing
```

cloudquery.log
```
{"level":"info","fileLogging":true,"jsonLogOutput":false,"consoleLog":true,"verbose":false,"logDirectory":".","fileName":"cloudquery.log","maxSizeMB":30,"maxBackups":3,"maxAgeInDays":3,"time":"2021-06-08T11:52:35-04:00","message":"logging configured"}
{"level":"info","time":"2021-06-08T11:52:35-04:00","message":"Initializing required providers"}
{"level":"info","name":"azure","version":"latest","time":"2021-06-08T11:52:35-04:00","message":"Initializing provider"}
{"level":"info","provider":"azure","version":"v0.2.1","time":"2021-06-08T11:52:35-04:00","message":"requesting provider to configure"}
{"level":"info","@module":"azure","subscriptions":["<SNIP>"],"timestamp":"2021-06-08T11:52:36.486-0400","time":"2021-06-08T11:52:36-04:00","message":"No subscriptions specified going to using all available ones"}
{"level":"info","provider":"azure","version":"v0.2.1","time":"2021-06-08T11:52:36-04:00","message":"provider configured successfully"}
{"level":"info","provider":"azure","version":"v0.2.1","time":"2021-06-08T11:52:36-04:00","message":"provider started fetching resources"}
{"level":"info","@module":"azure","count":0,"subscription_id":"<SNIP>","table":"azure_keyvault_vaults","timestamp":"2021-06-08T11:52:36.559-0400","time":"2021-06-08T11:52:36-04:00","message":"fetched successfully"}
{"level":"info","@module":"azure","count":0,"subscription_id":"<SNIP>","table":"azure_mysql_servers","timestamp":"2021-06-08T11:52:36.608-0400","time":"2021-06-08T11:52:36-04:00","message":"fetched successfully"}
{"level":"info","@module":"azure","count":0,"subscription_id":"<SNIP>","table":"azure_postgresql_servers","timestamp":"2021-06-08T11:52:36.660-0400","time":"2021-06-08T11:52:36-04:00","message":"fetched successfully"}
{"level":"info","@module":"azure","count":0,"subscription_id":"<SNIP>","table":"azure_sql_servers","timestamp":"2021-06-08T11:52:36.715-0400","time":"2021-06-08T11:52:36-04:00","message":"fetched successfully"}
{"level":"info","@module":"azure","count":2,"subscription_id":"<SNIP>","table":"azure_resource_groups","timestamp":"2021-06-08T11:52:36.750-0400","time":"2021-06-08T11:52:36-04:00","message":"fetched successfully"}
{"level":"info","@module":"azure","count":0,"subscription_id":"<SNIP>","table":"azure_storage_accounts","timestamp":"2021-06-08T11:52:36.750-0400","time":"2021-06-08T11:52:36-04:00","message":"fetched successfully"}
{"level":"info","@module":"azure","count":1,"subscription_id":"<SNIP>","table":"azure_compute_disks","timestamp":"2021-06-08T11:52:36.793-0400","time":"2021-06-08T11:52:36-04:00","message":"fetched successfully"}
```

**Comment by roneli** [comment]
Perfect, I assumed it was some kind of panic, will fix soon.  @jbertman thanks for the logs!

**Comment by roneli** [comment]
Fixed in [v0.2.2](https://github.com/cloudquery/cq-provider-azure/releases/tag/v0.2.2) update your hcl to either point to "latest" azure provider or "v0.2.2". i'll keep this issue open for now.

**Comment by jbertman** [comment]
Works great! Feel free to close if there is no further action required.
